client/my-sites/controller.js

```diff
@@ -269,7 +270,8 @@ module.exports = {
 		const siteFragment = context.params.site || route.getSiteFragment( context.path );
 		const basePath = route.sectionify( context.path );
 		const currentUser = user.get();
-		const hasOneSite = currentUser.visible_site_count === 1;
+		const visibleSiteCount = getVisibleSites( getState() ).length;
```

Comment:
Calculating the visible site count by filtering the current sites works to get the expected count. However, we still have the pesky problem that we can't trust `currentUser.visible_site_count` to be legit. 

Ideally, I think that whatever is causing the `currentUser.visible_site_count` to be incorrect on the backend should be fixed.

What are your thoughts on setting `currentUser.visible_site_count` to the results of `getVisibleSites( getState() ).length`?